### DISCUSSION TITLE
Version 2.16.0 error on Python 3.12

### DISCUSSION THREAD

**Discussion by rschirin** [jertel/elastalert2#1387] [Q&A] [answered]
hey there, I tried to update the used version, from v2.12.0 to v2.16.0. Unfortunately I am facing the following issue:

```
Reading Elastic 8 index mappings:
Reading index mapping 'es_mappings/8/silence.json'
Reading index mapping 'es_mappings/8/elastalert_status.json'
Reading index mapping 'es_mappings/8/elastalert.json'
Reading index mapping 'es_mappings/8/past_elastalert.json'
Reading index mapping 'es_mappings/8/elastalert_error.json'
Index elastalert_status already exists. Skipping index creation.
^[[ATraceback (most recent call last):
  File "/usr/local/bin/elastalert", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.12/site-packages/elastalert/elastalert.py", line 1923, in main
    client.start()
  File "/usr/local/lib/python3.12/site-packages/elastalert/elastalert.py", line 1147, in start
    self.scheduler.start()
  File "/usr/local/lib/python3.12/site-packages/apscheduler/schedulers/background.py", line 38, in start
    self._thread.start()
  File "/usr/local/lib/python3.12/threading.py", line 992, in start
    _start_new_thread(self._bootstrap, ())
RuntimeError: can't start new thread
```

Elasticsearch (v8.x) endpoint has TLS enabled but I have disabled the check into config.yml file.
Issue seems to start from v2.13.0 with the same error but referred to python3.11.

**Comment by jertel** [comment]
Another discussion also reported it but never followed-up. Unfortunately I'm unable to reproduce it. Can you reproduce it via the official Docker container? If so then perhaps with your rule and config files I might be able to finally reproduce it.

**Reply by rschirin** [reply]
I saw the other discussion right now. 
Try to add some details:
OS: debian 10
Docker version 18.09.1, build 4c52b90
I am using the official Elastalert2 Docker container.

I tried to reproduce it locally but I cannot face the error, probably it is something related to arch? 
I will work on it

**Reply by rschirin** [reply]
Ok, this is the report:

debian 10: not works
debian 12: works

interesting scenario. To be honest, I don't know if I will able to find the reason of failure on Debian 10.

**Reply by jertel** [reply, marked as answer]
Thanks. It seems that running a Debian 12 container on a Debian 10 host is problematic, and leads to the "can't start new thread" error. Given that Debian 10's end-of-life was in 2022, and the LTS 10's EOL is in a few months, you probably will need to stick with the older ElastAlert 2 version for now, and invest the time on getting your OS upgraded.

**Reply by rschirin** [reply]
Absolutely fine. I have already speed up my road map to update OS. 
Glad that everything is solved now
